### PR TITLE
Translation support

### LINKED ISSUE #2805: Support content translation
Add support for translating content/post paragraphs, start with support waves translations and then extend to full blog posts.

**Phase 1:**
let's make a screen/component where we can try out various options
try each options for a certain post... for the start, take english to spanis or spanish to english. 
and compare the output or each option we have
if offline variants are able to give us acceptable output
I think we do not need to go for paid option for the start

**Phase 2:**
Finalise the initial UI for waves translation

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
This PR adds support for content translation in waves screen

### Steps to reproduce
Go to Waves screen -> open post options from 3 dot menu -> tap on translate option
It would open a translation sheet with different languages selector and translated content

### Issue number
fixes #2805 

### Screenshots/Video

https://github.com/ecency/ecency-mobile/assets/48380998/f4c8b7ab-4d10-4f2a-a685-3c15d3f0cb5d

### CODE REVIEW

**Review by noumantahir — CHANGES_REQUESTED** [review]
- add better support for light and dark themes
- modal goes too up high, either make it dynamic based on content length, or make sure it does not go behind notch
- may be rearrange post options items so translate item it little more visible

Screenshot 2024-01-16 at 14 48 53
Screenshot 2024-01-16 at 14 48 09

**Comment by aliseyalvi** [comment]
image
image

File: src/components/post-translation-modal/postTranslationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    setTranslatedPost('');
+    setTranslationError('');
+    setSelectedSourceLang(srcLang);
+    setSelectedTargetLang(targetLang);
+  };
+
+  const _renderLanguageSelector = () => (
+    <View style={styles.languageSelectorRow}>
+      <View style={styles.row}>
+        <Text style={styles.labelText}>From</Text>
```
Comment (noumantahir) [inline src/components/post-translation-modal/postTranslationModal.tsx:111]: use intl

File: src/components/post-translation-modal/postTranslationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+          buttonStyle={styles.dropdownBtnStyle}
+          buttonTextStyle={styles.dropdownBtnTextStyle}
+          rowTextStyle={styles.dropdownRowTextStyle}
+          selectedRowStyle={styles.dropdownSelectedRowStyle}
+          selectedRowTextStyle={styles.dropdownSelectedRowTextStyle}
+          dropdownFlatlistProps={scrollHandlers}
+        />
+      </View>
+      <View style={styles.row}>
+        <Text style={[styles.labelText, styles.toText]}>To</Text>
```
Comment (noumantahir) [inline src/components/post-translation-modal/postTranslationModal.tsx:134]: use intl

File: src/components/post-translation-modal/postTranslationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        <SelectDropdown
+          data={supportedLangsList.filter(
+            (item) => item.code !== srcLang.code || item.code !== selectedSourceLang.code,
```
Comment (noumantahir) [inline src/components/post-translation-modal/postTranslationModal.tsx:137]: possible to default `To` language to selected app language?

**Review by noumantahir — CHANGES_REQUESTED** [review]
overall good, two tiny change and one conditional change requested

**Comment by aliseyalvi** [comment]
@noumantahir 

1. added intl
2. added support for auto select target lang according to app default language

https://github.com/ecency/ecency-mobile/assets/48380998/b18a05c9-626a-48c0-8479-122ff6cd93c9

**Review by noumantahir — APPROVED** [review]
functionality wise it's smooth and very useable, user experience can get better as we iterate later.

**Pull request merged**